x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/dsl_queries.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { dsl as genericDsl } from '../../common/data/dsl_queries';
+
+export const dsl = {
+  isInstalled(): QueryDslQueryContainer {
+    return { exists: { field: 'elastic_rule.id' } };
+  },
+  isNotInstalled(): QueryDslQueryContainer {
+    return { bool: { must_not: dsl.isInstalled() } };
+  },
+  isPrebuilt(): QueryDslQueryContainer {
+    return { exists: { field: 'elastic_rule.prebuilt_rule_id' } };
+  },
+  isCustom(): QueryDslQueryContainer {
+    return { bool: { must_not: dsl.isPrebuilt() } };
+  },
+  matchTitle(title: string): QueryDslQueryContainer {
+    return { match: { 'elastic_rule.title': title } };
+  },
+  isInstallable(): QueryDslQueryContainer[] {
+    return [genericDsl.isFullyTranslated(), dsl.isNotInstalled()];
+  },
+  isNotInstallable(): QueryDslQueryContainer[] {
+    return [genericDsl.isNotFullyTranslated(), dsl.isInstalled()];
+  },
+  isMissingIndex: genericDsl.isMissingIndex,
```

Comment:
Why do we need to add this here? It does not seem to be used by the rules `getFilterQuery`, only in the test.

Reply:
It is now changes. we define `isMissingIndex` here only, instead of importing it.